enhancements/insights/on-demand-data-gathering.md

```diff
@@ -0,0 +1,340 @@
+// +openshift:compatibility-gen:level=4
+type DataGatherList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []DataGather `json:"items"`
+}
+```
+
+### Workflow Description
+
+1. User/cluster admin creates a new CR instance of the `datagather.insights.openshift.io` CRD. 
+2. The Insights operator is informed (using informers) about the new CR instance and spawns a new data gathering in a new container. 
+3. The new container process gathers Insights data based on the configuration provided in the CR instance and is responsible for uploading the gathered Insights archive and checking that the archive was processed correctly in the console.redhat.com (the API documentation is available at https://console.redhat.com/docs/api/insights-results-aggregator/v2). 
+4. The original Insights operator's pod is responsible for downloading the last known Insights analysis report (from console.redhat.com)
+   based on latest successfully finished gathering job and updating the corresponding values in the already existing `insightsoperator.operator.openshift.io` resource.
+
+This scenario brings few questions and things to be discussed. Please see the following sections [Implementation details](#implementation-detailsnotesconstraints), [Risk mitigations](#risks-and-mitigations) and [Open questions](#open-questions)
+
+### API Extensions
+
+This is adding new Insights CRD in the `insights.openshift.io/v1alpha1` . The proposal of the structure of the new CRD is mentioned in the [Proposal](#Proposal) section.
+
+### Implementation Details/Notes/Constraints
+
+#### Current behavior
+The Insights operator currently runs and gathers the data in a single container. 
+Uploading of new archives, pruning old archives as well as downloading the last Insights analysis report (this is always per archive) is implemented in this one container/binary. 
+When there is a successful upload (to the console.redhat.com) of the archive then there is some short delay (waiting for the processing in console.redhat.com pipelines - see the [API documentation](https://console.redhat.com/docs/api/insights-results-aggregator/v2)) and then there is a request to get the last Insights analysis report per given cluster. 
+Finally the status section of the `insightsoperator.operator.openshift.io` CR is updated. Note that there is a retry logic in case of failed upload and after some time (and number of attempts) the Insights operator is marked as degraded.
+
+#### Updates required to allow on-demand data gathering
+The plan is to refactor the existing periodic data gathering to run in a separate Pod/Job. The existing Insights operator pod will still exist, but it would periodically do the following:
+- check the `insightsdatagather.config.openshift.io` CR to read current user configuration
+- create a new `datagather.insights.openshift.io` CR (in the `openshift-insights` namespace) based on the values from the above
```

Comment:
why is the resource namespaced?  What happens if there's an instance in another namespace?

Reply:
I think this is slightly related to the https://github.com/openshift/enhancements/pull/1279#discussion_r1088407872. I will enhance the description and reasoning, but I believe we only want to watch the new CRS in the `openshift-insights` namespace and if a new CR is created in another namespace, nothing happens.

Reply:
I mentioned this in [Proposal](https://github.com/openshift/enhancements/blob/40d7a2d2fe4eb44d0cbf96e9bf4770bceec95a12/enhancements/insights/on-demand-data-gathering.md#proposal) section.